torch/csrc/jit/frontend/concrete_module_type.cpp

```diff
@@ -186,6 +186,10 @@ c10::optional<std::string> ConcreteModuleType::findFailedAttribute(
   return c10::nullopt;
 }
 
+bool ConcreteModuleType::isIgnoredAttribute(const std::string& name) const {
```

Comment:
You need to edit `ConcreteModuleTypeBuilder::equals` so that this field gets used to disambiguate instances of a concrete type.

Reply:
I added this but it's hard to test because ignoring an attribute changes the list of attributes on the `ConcreteModuleTypeBuilder` instance. The equality check between `ConcreteModuleTypeBuilder` instances (also `ConcreteModuleType` and `Builder` instances) fails when when an attribute is ignored even if there is no `ignoredAttributes_ == other.ignoredAttributes_` condition added to `ConcreteModuleTypeBuilder::equals` because `attributes_ == other.attributes_` is `false` . 

I'm not saying we shouldn't have it, but I can't prove that it works.

Reply:
As far as I understand, ConcreteMuduleType::equals is to disambiguate between different instantiations of an nn.Module which can result in a different type. `__jit_ignored_attributes__` is a class-level attribute, I don't see how it could differ between different instantiations.

Reply:
That's my understanding too, but as per one of my tests, the class attribute can be edited between invocations to `torch.jit.script`. I don't think it's a realistic use case, but it is possible.

Reply:
It can be edited, or we can set it in the body of the `__init__`, e.g. 
```
def __init__(self, ignored):
    self.__ignored__ = ignored
```
Both happen with some degree of regularity for other elements of ConcreteModuleType. But good point that it changes the available attributes, so checking ignored attrs is somewhat redundant. I think we should do it still for consistency, since all other attributes of ConcreteModuleType should participate in equality.